### ISSUE TITLE
Implementing event passthrough for integration.

### ISSUE THREAD

**Issue by mobile-bungalow** [hecrj/iced#464]
The `Program` trait and `program::State` struct used in the [integration demo](https://github.com/hecrj/iced/tree/master/examples/integration) expose no method for determining whether events passed to the program are intercepted by the UI. 

This is a feature that could be useful for any integrated application that uses Iced as an overlay above another piece of Interactive UI. e.g. games, CAD software. 

is this feature in scope for Iced? If so what steps need to happen before trying to implement it?

**Comment by mobile-bungalow** [comment]
duplicate of #408
